src/pyhf/modifiers/lumi.py

```diff
@@ -69,4 +69,8 @@ def apply(self, pars):
         else:
             results_lumi = tensorlib.einsum('msab,xa->msab', self.lumi_mask, lumis)
 
-        return tensorlib.where(self.lumi_mask, results_lumi, self.lumi_default)
+        return tensorlib.where(
+            tensorlib.astensor(self.lumi_mask, dtype="bool"),
```

Comment:
this can probably go in the _precompute of the modifier where `self.lumi_mask` is set up

Reply:
@lukasheinrich I looked at this again and I don't think this will work. If `einsum` is given `self.lumi_mask` as a Bool tensor then it will crash as the backends don't support `Bool` there. Sorry, I ran into this last night but didn't mention it anywhere.

Reply:
ok I guess we could have `self.lumi_mask_bool`, and `self.lumi_mask` and use each at their appropriate position, but probably doesn't make that much of a diff

Reply:
I can do this.